integration/management-api-tests/src/test/java/org/wso2/micro/integrator/api/ConfigsResourceTestCase.java

```diff
@@ -0,0 +1,270 @@
+        String accessToken = TokenUtil.getAccessToken(hostName, offset);
+        Assert.assertNotNull(accessToken);
+
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Accept", "application/json");
+        headers.put("Authorization", "Bearer " + accessToken);
+
+        String endpoint = "https://" + hostName + ":" + (DEFAULT_INTERNAL_API_HTTPS_PORT + offset) + "/management/"
+                + "configs?configName=correlation";
+
+        SimpleHttpClient client = new SimpleHttpClient();
+
+        HttpResponse response = client.doGet(endpoint, headers);
+        String responsePayload = client.getResponsePayload(response);
+        Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
+        JSONObject jsonResponse = new JSONObject(responsePayload);
+        Assert.assertTrue(jsonResponse.has("configName"));
+        Assert.assertEquals(jsonResponse.get("configName"), "correlation");
+        Assert.assertTrue(jsonResponse.has("configs"));
+        JSONObject responseConfigs = (JSONObject) jsonResponse.get("configs");
+        Assert.assertTrue(responseConfigs.has("enabled"));
+        Assert.assertEquals(responseConfigs.get("enabled"), correlationConfigStatus);
+    }
+
+
+    private void sendPutRequest(int offset, boolean correlationConfigEnabled) throws IOException {
+        String accessToken = TokenUtil.getAccessToken(hostName, offset);
+        Assert.assertNotNull(accessToken);
+
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Accept", "application/json");
+        headers.put("Authorization", "Bearer " + accessToken);
+
+        String endpoint = "https://" + hostName + ":" + (DEFAULT_INTERNAL_API_HTTPS_PORT + offset) + "/management/"
+                + "configs";
+
+        JSONObject payloadConfigs = new JSONObject();
+        payloadConfigs.put("enabled", correlationConfigEnabled);
+        JSONObject payload = new JSONObject();
+        payload.put("configName", "correlation");
+        payload.put("configs", payloadConfigs);
+
+
+        SimpleHttpClient client = new SimpleHttpClient();
+        HttpResponse response = client.doPut(endpoint, headers, payload.toString(), "application/json");
+        String responsePayload = client.getResponsePayload(response);
+        Assert.assertEquals(response.getStatusLine().getStatusCode(), 200);
+        JSONObject jsonResponse = new JSONObject(responsePayload);
+        Assert.assertTrue(jsonResponse.has("message"));
+        Assert.assertEquals(jsonResponse.get("message"), "Successfully Updated Correlation Logs Status");
+    }
+
+    /**
+     * Return if the management api is available.
+     * @return
+     */
+    public Callable<Boolean> isManagementApiAvailable() {
+        return this::checkIfManagementApiAvailable;
+    }
+
+    /**
+     * return if the management api is unavailable.
+     *
+     * @return callable callable
+     */
+    public Callable<Boolean> isManagementApiUnAvailable() {
```

Comment:
If this method is not used, shall we remove it?

Reply:
Okay ayye